backend/graph/nodes.py

```
@@ -0,0 +1,58 @@
+from src.utils.rag import get_rag_system_prompt
+from .model import llm, llm_with_tools
+from .state import State
+
+
+def detect_language_node(state: State):
+    """Detect the language of the last user message."""
+    for msg in reversed(state.messages):
+        if isinstance(msg, HumanMessage) and isinstance(msg.content, str):
+            try:
+                detected = detect_language(msg.content)
+            except Exception:
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)